internal/statestore/redis.go

```diff
@@ -223,18 +223,22 @@ func redisURLFromAddr(addr string, cfg config.View, usePassword bool) string {
 func (rb *redisBackend) connect(ctx context.Context) (redis.Conn, error) {
 	redisConn, err := rb.redisPool.GetContext(ctx)
 	if err != nil {
-		redisLogger.WithFields(logrus.Fields{
-			"error": err.Error(),
-		}).Error("failed to connect to redis")
-		return nil, status.Errorf(codes.Unavailable, "%v", err)
+		return nil, status.Errorf(codes.Unavailable, "failed to connect to redis: %v", err)
 	}
 	return redisConn, nil
 }
 
 // CreateTicket creates a new Ticket in the state storage. If the id already exists, it will be overwritten.
 func (rb *redisBackend) CreateTicket(ctx context.Context, ticket *pb.Ticket) error {
+	if ticket == nil {
+		return status.Error(codes.InvalidArgument, "nil ticket provided")
+	}
+
 	redisConn, err := rb.connect(ctx)
 	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
```

Comment:
Per the linked issue: https://github.com/googleforgames/open-match/issues/1102

It's not good to both log an error and also return it, this is the cause of the initial issue.